wasm.md

```diff
@@ -1531,6 +1496,238 @@ Start Function
+ // ------------------------------------------------
+
+    syntax GlobalSpec ::= InlineExport GlobalSpec
+ // ---------------------------------------------
+    rule <k> ( global                  EXPO:InlineExport SPEC:GlobalSpec )
+          => ( global #freshId(NEXTID) EXPO              SPEC            )
+          ...
+         </k>
+         <nextFreshId> NEXTID => NEXTID +Int 1 </nextFreshId>
+
+    rule <k> ( global ID:Identifier ( export ENAME ) SPEC:GlobalSpec )
+          => ( export ENAME ( global ID ) )
+          ~> ( global ID SPEC )
+          ...
+         </k>
+
+    syntax FuncSpec   ::= InlineExport FuncSpec
+ // -------------------------------------------
+    rule <k> ( func                  EXPO:InlineExport SPEC:FuncSpec )
+          => ( func #freshId(NEXTID) EXPO              SPEC          )
+          ...
+         </k>
+         <nextFreshId> NEXTID => NEXTID +Int 1 </nextFreshId>
+
+    rule <k> ( func ID:Identifier ( export ENAME ) SPEC:FuncSpec )
+          => ( export ENAME ( func ID ) )
+          ~> ( func ID SPEC )
+          ...
+         </k>
+
+    syntax TableSpec  ::= InlineExport TableSpec
+ // --------------------------------------------
+    rule <k> ( table                  EXPO:InlineExport SPEC:TableSpec )
+          => ( table #freshId(NEXTID) EXPO              SPEC           )
+          ...
+         </k>
+         <nextFreshId> NEXTID => NEXTID +Int 1 </nextFreshId>
+
+    rule <k> ( table ID:Identifier ( export ENAME ) SPEC:TableSpec )
+          => ( export ENAME ( table ID ) )
+          ~> ( table ID SPEC )
+          ...
+         </k>
+
+    syntax MemorySpec ::= InlineExport MemorySpec
+ // ---------------------------------------------
+    rule <k> ( memory                  EXPO:InlineExport SPEC:MemorySpec )
+          => ( memory #freshId(NEXTID) EXPO              SPEC            )
+          ...
+         </k>
+         <nextFreshId> NEXTID => NEXTID +Int 1 </nextFreshId>
+
+    rule <k> ( memory ID:Identifier ( export ENAME ) SPEC:MemorySpec )
+          => ( export ENAME ( memory ID ) )
+          ~> ( memory ID SPEC )
+          ...
+         </k>
+```
+
+Imports
+-------
+
+Imports need to describe the type of what is imported.
+That an import is really a subtype of the declared import needs to be checked at instantiation time.
+The value of a global gets copied when it is imported.
+
+```k
+    syntax Defn       ::= ImportDefn
+    syntax ImportDefn ::= "(" "import" String String ImportDesc ")"
+    syntax ImportDesc ::= "(" "func"   OptionalId TypeUse        ")" [klabel(funcImportDesc)]
+                        | "(" "table"  OptionalId TableType      ")" [klabel( tabImportDesc)]
+                        | "(" "memory" OptionalId MemType        ")" [klabel( memImportDesc)]
+                        | "(" "global" OptionalId TextGlobalType ")" [klabel(globImportDesc)]
+ // -------------------------------------------------------------------------------------
```

Comment:
syntax here (----- line up with ])

Reply:
Also isn't it weird that we have `TableType`, `MemType`, but `TextGlobalType`? Can we either make them all with `Text` or all without `Text`?

Reply:
Yeah, I don't think it's super elegant either, but it's the best I could come up with.

The problem is that this is one of the places where abstract and concrete syntax diverge. `TextGlobalType` is concrete, and `GlobalType` is abstract. The abstract syntax annotates every global with `var` or `const`, but the concrete just puts `mut` on the mutable and nothing on the constant values. So if we want to use `var` and `const` in our rules (which I think we should), we need to differentiate them from the text syntax (which we need to parse) which just uses `mut`. For the other types the abstract and constant syntaxes are the same, so naming them `Text*` would be strange (since it's not specifically concrete syntax).

Reply:
How about:
TextGlobalType -> GlobalType
GlobalType -> AbstractGlobalType?
This looks a little easier to understand (and more consistent) to me.

Reply:
Or you can use `TextFormatGlobalType`, this looks a little better than `TextGlobalType`

Reply:
Okay, `TextFormatGlobalType` makes sense to me.

Also, I just realized that in the spec, all these productions have the same name, e.g., there are two productions called `globaltype`, one for the text format and one for the abstract syntax, and the only difference is the font. So we should maybe adopt some convention for differentiating them. I think `TextFormat*` becomes a bit long if we use it in many places, so maybe, `TF*` as a prefix where necessary would be better?